src/main.ts

```diff
@@ -40,9 +42,24 @@ const init = async (): Promise<void> => {
             forceGraphQLImport: false,
             skipGraphQLImport: true,
         });
+        const schema = makeExecutableSchema({ typeDefs, resolvers });
         const apolloServer = new ApolloServer({
-            typeDefs,
-            resolvers,
+            schema,
+            validationRules: [
+                depthLimit(5),
+                // may need to have resolver level complexity, but this is okay for now.
+                queryComplexity({
+                    maximumComplexity: 1000,
+                    estimators: [
+                        // default fallback estimator.
+                        simpleEstimator({
+                            defaultComplexity: 1,
+                        }),
+                    ],
+                }),
+            ],
+            // @todo: Introspection queries will be blocked unless you are authenticated.
```

Comment:
Can we make this switchable for Production? What do others think @will-byrne @erezmus @Cooryd

Reply:
Sure